src/orc.jl

```diff
@@ -0,0 +1,151 @@
+export OrcJIT, OrcModule, OrcTargetAddress
+export JITEventListener, GDBRegistrationListener, IntelJITEventListener,
+       OProfileJITEventListener, PerfJITEventListener
+export dispose, errormsg, compile!, remove!, add!,
+       mangle, address, addressin, create_stub!, set_stub!,
+       register!, unregister!, callback!
+export JITTargetMachine
+
+@checked struct OrcJIT 
+    ref::API.LLVMOrcJITStackRef
+end
+
+Base.unsafe_convert(::Type{API.LLVMOrcJITStackRef}, orc::OrcJIT) = orc.ref
+
+"""
+    OrcJIT(::TargetMachine)
+
+Creates a OrcJIT stack based on the provided target machine.
+
+!!! warning
+    Takes ownership of the provided target machine.
+"""
+function OrcJIT(tm::TargetMachine)
+    OrcJIT(API.LLVMOrcCreateInstance(tm))
+end
+
+function dispose(orc::OrcJIT)
+    API.LLVMOrcDisposeInstance(orc)
+end
+
+function errormsg(orc::OrcJIT)
+    unsafe_string(LLVM.API.LLVMOrcGetErrorMsg(orc))
```

Comment:
Don't you need to dispose this message afterwards? If so, we have `unsafe_message`

Reply:
To quote from the doc string

>  * This message is owned by the ORC JIT Stack and will be freed when the stack
> * is disposed of by LLVMOrcDisposeInstance.